Doc #128 including images in exercises (#612)
ocaml-sf/learn-ocaml: docs/howto-include_images_in_exercises.md

> How to include static content in an Exercise Statement
> ======================================================

Reviewer: Maybe this should be rephrased to match the filename/url ?
```suggestion
How to include images in exercises
==================================
```

BTW, please rename the .md so that it only has dashes:

```
git mv docs/howto-include_images_in_exercises.md docs/howto-include-images-in-exercises.md
git commit -m "Rename"
```

> How to include our own static content to an Exercise Statement
> ==============================================================
> 
> ## Context
> 
> If you create an exercise statement, you probably want to add your own static content such as images in an exercise statement. 
> 
> `LearnOCaml` provides a feature to include static contents in an exercise statement. In addition to always being able to include a link to external content, it's possible to include specific static content. 
> 
> ## How-to
> 
> Lets see how to include `OCaml.png` in exercise `demo`.
> Follow these steps:
> 
> 1. Create a directory `<repo>/exercices/demo/subdirectory`. (The name of the `subdirectory` is free.)
> 2. Put `OCaml.png` in there.
> 3. In `<repo>/exercises/demo/decr.html` or `<repo>/exercises/demo/descr.md`, use the following syntaxe <br> `<img src="/static/demo/subdirectory/OCaml.png">`. 
> 
> To do this we can't use relative URLs. 

Reviewer: Thank you @CJs0800 for your PR !

LGTM, I'll just suggest a few minor changes to your English documentation:

```suggestion
How to include static content in an Exercise Statement
======================================================

## Context

If you create an exercise, you may want to add your own static content such as images in the exercise description (`descr.html` or `descr.md`).

Learn-OCaml provides this feature, so that you can either add images hosted in an external domain, or images statically served by the Learn-OCaml server itself (which can be handy for summative exams, for example, when remote resources are blocked).

## How-to

In order to include `OCaml.png` in an exercise called `demo`, you just need to:

1. Create a directory `<repo>/exercices/demo/subdirectory`. (The name of the `subdirectory` is free.)
2. Put `OCaml.png` in there.
3. In `<repo>/exercises/demo/descr.html` (resp. `<repo>/exercises/demo/descr.md`), use the following syntax:  
`<img src="/static/demo/subdirectory/OCaml.png" alt="fallback text">` (resp. `![fallback text](/static/demo/subdirectory/OCaml.png)`).

Note that we can't use relative URLs, hence the leading `/`.
```

> Note that we can't use relative URLs, hence the leading `/`.

Reviewer: And you can add some final advice about the images size:

```suggestion
Note that we can't use relative URLs, hence the leading `/`.

If you want to fix the width (or height) of an image in `descr.html` or `descr.md`, you can use the usual HTML syntax:  
`<img src="/static/demo/subdirectory/OCaml.png" alt="fallback text width="400px"/>`
```
